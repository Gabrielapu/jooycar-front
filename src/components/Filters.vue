<template>
  <div class="filters">
    <DatePicker 
      v-model="startDate" 
      class="start-picker ml-4"
      placeholder="Fecha de Inicio"
      @clear="startDate = null; emitFilters()"
    />
    <DatePicker 
      v-model="endDate" 
      class="end-picker"
      placeholder="Fecha de inicio"
      :min-date="startDate"
      @clear="endDate = null; emitFilters()"
    />
    <v-text-field
      v-model="distance"
      label="Distancia"
      class="distance-input mr-2"
      variant="outlined"
      density="compact"
      clearable
      hide-details
      type="number"
      min="0"
      @click:clear="distance = null; emitFilters()"
    />
    <v-btn
      color="primary"
      variant="outlined"
      class="search-button mr-4"
      @click="emitFilters()"
    >
      Buscar
    </v-btn>
  </div>
</template>

<script setup>
import { ref } from 'vue';
import DatePicker from './DatePicker.vue';

const emit = defineEmits(['search'])
const startDate = ref(null)
const endDate = ref(null)
const distance = ref(null)

function emitFilters() {
  emit('search', {
    startDate: Date.parse(startDate.value),
    endDate: Date.parse(endDate.value),
    distance: distance.value
  })
}
</script>