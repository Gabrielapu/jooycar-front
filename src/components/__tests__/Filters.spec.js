import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import Filters from '@/components/Filters.vue'  

describe('Filters.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(Filters)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
