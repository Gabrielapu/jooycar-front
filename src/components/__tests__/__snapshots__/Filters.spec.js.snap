// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filters.vue > renders correctly 1`] = `
"<div class="filters">
  <v-menu>
    <v-date-picker header="" title="Seleccionar fecha"></v-date-picker>
  </v-menu>
  <v-menu>
    <v-date-picker header="" title="Seleccionar fecha"></v-date-picker>
  </v-menu>
  <v-text-field label="Distancia" class="mr-2" variant="outlined" density="compact" clearable="" hide-details="" type="number" min="0"></v-text-field>
  <v-btn color="primary" variant="outlined" class="mr-4"> Buscar </v-btn>
</div>"
`;
